### PR TITLE
support auth mechanism

### PR DESCRIPTION
**Pull request by Yancey0623**
second of #49

### CODE REVIEW

File: driver_test.go [hunk 1]
```diff
@@ -50,6 +51,7 @@ func TestColumnTypeName(t *testing.T) {
 	a := assert.New(t)
 	db, _ := sql.Open("hive", "127.0.0.1:10000/churn")
 	rows, err := db.Query("SELECT customerID, gender FROM train")
+	fmt.Println(err)
```
Comment (tonyyang-svail) [inline driver_test.go]: clean up.

**Review by tonyyang-svail — COMMENTED** [review]
Is it possible to test `auth` against HIVE server?

File: driver_test.go (hunk 1 above)
Comment (Yancey0623) [reply driver_test.go]: done.

**Comment by Yancey0623** [comment]
> Is it possible to test auth against HIVE server?

@tonyyang-svail add a unit test to against the auth.

**Review by tonyyang-svail — APPROVED** [review]
LGTM

**Pull request merged**